test/test_fcntl.rs

```diff
@@ -561,3 +561,31 @@ mod test_posix_fallocate {
         }
     }
 }
+
+#[cfg(any(
+    target_os = "dragonfly",
+    target_os = "netbsd",
+    target_os = "macos",
+    target_os = "ios"
+))]
+mod test_apple_netbsd {
+    use nix::fcntl::*;
+    use std::os::unix::io::AsRawFd;
+    use std::path::PathBuf;
+    use tempfile::NamedTempFile;
+
+    #[test]
+    fn test_path() {
+        let tmp = NamedTempFile::new().unwrap();
+        let fd = tmp.as_raw_fd();
+        let mut path = PathBuf::new();
+        let res =
+            fcntl(fd, FcntlArg::F_GETPATH(&mut path)).expect("get path failed");
+        assert_ne!(res, -1);
+        // We purposely check the filenames, since they are supposed to be unique.
+        // The left operand tend to have `/private/var` whereas the right does not contains
+        // the `/private` part even tough in practice they lead to the same file, indeed
+        // `/var` is a symlink to `/private/var`.
```

Comment:
This comment is confusing.  Are you saying that there are some symlinks in one path or the other?  If so, I suggest just using `Path::canonicalize` on both sizes.

Reply:
The code is better, but the comment is still confusing, especially because it only applies to OSX.  With the canonicalize in place, I don't think the comment is even necessary.

Reply:
Resolve this as the comment has been removed